test/metabase/upload_test.clj

```diff
@@ -539,387 +598,372 @@
-                           "18,0"]))
-          (testing "Table and Fields exist after sync"
-            (sync/sync-database! (mt/db))
-            (let [table (t2/select-one Table :db_id (mt/id))]
-              (is (=? {:name #"(?i)upload_test"} table))
-              (testing "Check the boolean column has a boolean base_type"
-                (is (=? {:name      #"(?i)bool"
-                         :base_type :type/Boolean}
-                        (t2/select-one Field :database_position 2 :table_id (:id table)))))
-              (testing "Check the data was uploaded into the table correctly"
-                (let [bool-column (map #(nth % 2) (rows-for-table table))
-                      alternating (map even? (range (count bool-column)))]
-                  (is (= alternating bool-column)))))))))))
+          (let [table-name (mt/random-name)]
+            (@#'upload/load-from-csv!
+             driver/*driver*
+             (mt/id)
+             table-name
+             (csv-file-with ["id,bool"
+                             "1,true"
+                             "2,false"
+                             "3,TRUE"
+                             "4,FALSE"
+                             "5,t    "
+                             "6,   f"
+                             "7,\tT"
+                             "8,F\t"
+                             "9,y"
+                             "10,n"
+                             "11,Y"
+                             "12,N"
+                             "13,yes"
+                             "14,no"
+                             "15,YES"
+                             "16,NO"
+                             "17,1"
+                             "18,0"]))
+            (testing "Table and Fields exist after sync"
+              (let [table (sync-upload-test-table! :database (mt/db) :table-name table-name)]
+                (testing "Check the boolean column has a boolean base_type"
+                  (is (=? {:name      #"(?i)bool"
+                           :base_type :type/Boolean}
+                          (t2/select-one Field :database_position 2 :table_id (:id table)))))
+                (testing "Check the data was uploaded into the table correctly"
+                  (let [bool-column (map #(nth % 2) (rows-for-table table))
+                        alternating (map even? (range (count bool-column)))]
+                    (is (= alternating bool-column))))))))))))
 
 (deftest load-from-csv-length-test
   (testing "Upload a CSV file with large names and numbers"
     (mt/test-drivers (mt/normal-drivers-with-feature :uploads)
       (let [length-limit (driver/table-name-length-limit driver/*driver*)
-            long-name    (apply str (repeat 33 "abcdefgh")) ; 33×8 = 264. Max is H2 at 256
-            short-name   (subs long-name 0 (- length-limit (count "_yyyyMMddHHmmss")))]
+            long-name    (apply str (shuffle (apply concat (repeat 33 (seq "abcdefgh"))))) ; 33×8 = 264. Max is H2 at 256
```

Comment:
I'm curious why we need this shuffle here, is it to work around flakes with `unique-table-name` again?

Reply:
It's to work around the fact that `mt/with-empty-db` doesn't clean up after itself with redshift. Instead there is one database for the entire test run, and a schema is created for the session during initialization. See [this](https://github.com/metabase/metabase/blob/99e3e107ebb8b2c5265682f6c311fd9e82237d7c/modules/drivers/redshift/test/metabase/test/data/redshift.clj#L132-L134). The table remains in the test database and the unused tables are cleaned up in regular intervals. I could probably fix this by cleaning up the table explicitly in each test run, but opted for this workaround.

Reply:
A short comment to the effect would be nice

```clojure
;; Ensure the name is unique, as `with-empty-db` unfortunately leaks tables when using redshift
```

This whole "Solve for N in Nx8" mechanism is also a bit overly complex IMHO, how about something like ` (->> (random-uuid) str cycle (take 254) (apply str))`?

Reply:
Actually it will still be necessary after https://github.com/metabase/metabase/pull/38453 so I've cleaned it up here 854ab1f51131ee601d62fac0fb1d3cc2dcd0e8eb